client/java-armeria/src/main/java/com/linecorp/centraldogma/client/armeria/AbstractArmeriaCentralDogmaBuilder.java

```diff
@@ -164,15 +163,18 @@ protected final EndpointGroup endpointGroup() throws UnknownHostException {
             group = new CompositeEndpointGroup(groups, EndpointSelectionStrategy.roundRobin());
         }
 
-        if (!healthCheckInterval.isZero()) {
-            return HealthCheckedEndpointGroup.builder(group, HttpApiV1Constants.HEALTH_CHECK_PATH)
-                                             .clientFactory(clientFactory)
-                                             .protocol(isUseTls() ? SessionProtocol.HTTPS
-                                                                  : SessionProtocol.HTTP)
-                                             .retryInterval(healthCheckInterval)
-                                             .build();
-        } else {
+        if (healthCheckInterval != null && healthCheckInterval.isZero()) {
             return group;
+        } else {
+            final HealthCheckedEndpointGroupBuilder healthCheckedEndpointGroupBuilder =
```

Comment:
This is not your fault but how about also applying the `HealthCheckedEndpointGroup` which is created at line 138?